stixcore/products/levelb/binary.py

```diff
@@ -283,6 +252,73 @@ def from_tm(cls, tmfile):
                              ssid=ssid, control=control, data=data)
             yield product
 
+    def extract_sequences(self):
+        """
+        Extract complete and incomplete packet sequences.
+        Returns
+        -------
+        tuple
+            LevelB products for the complete and incomplete sequences
+        """
+        if self.service_type == 3 and self.service_subtype == 25 and self.ssid in {1, 2}:
+            return [self], []
+        elif self.service_type == 21 and self.service_subtype == 6 and self.ssid in {30, 31, 32,
+                                                                                     33, 34, 43}:
+            return [self], []
+
+        sequences = []
+        flags = self.control['sequence_flag']
+        cur_seq = None
+        for i, f in enumerate(flags):
+            if f == SequenceFlag.STANDALONE:
+                sequences.append([i])
+            elif f == SequenceFlag.FIRST:
+                cur_seq = [i]
+
+            if f == SequenceFlag.MIDDLE:
+                if cur_seq:
+                    cur_seq.append(i)
+                else:
+                    cur_seq = [i]
+
+            if f == SequenceFlag.LAST:
+                if cur_seq:
+                    cur_seq.append(i)
+                    sequences.append(cur_seq)
+                else:
+                    sequences.append([i])
+
+                cur_seq = None
+
+        if cur_seq:
+            sequences.append(cur_seq)
+
+        complete = []
+        incomplete = []
+
+        for seq in sequences:
+            if len(seq) == 1 and flags[seq[0]] == SequenceFlag.STANDALONE:
+                complete.append(self[seq])
+            elif (flags[seq[0]] == SequenceFlag.FIRST
+                  and flags[seq[-1]] == SequenceFlag.LAST
+                  and (((self.control['sequence_count'][seq[-1]]
+                         + (  # possible role over off 14bit psc value
+                            0 if self.control['sequence_count'][seq[-1]] >
+                            self.control['sequence_count'][seq[0]] else
+                            max(self.control['sequence_count'][seq]))) -
+                        self.control['sequence_count'][seq[0]] + 1) >= len(seq) - 1)
+                    # -1 allow one gap in the sequence
```

Comment:
Is this defined in the FSW, I mean could the gap be larger or was this just based on what was seen in the TM?

Reply:
yes you are right this is just based on "observation of our data": never seen something else SO FARE.
i have no better idea at the moment